engines/router/README.md

```diff
@@ -74,3 +74,4 @@ __Note:__ The app uses the `in_forward` plugin to write data to a TCP socket.
     -H "X-B3-Traceid: 950f2de0b8430e9fa30ec88c39471716" \
     -d '{}'
 ```
+4. Grpc routes needs to be serving with reflection on, in order for UPI router to initialise successfully.
```

Comment:
For UPI no, since the contract will be known and they are not called by Fiber. This behavior is as such because Fiber has no knowledge of UPI and uses reflection to get the response proto.

It should be part of Turing service definition later on with the following signature.

```
service RouterService {
  rpc Enrich(PredictValuesRequest) returns (PredictValuesRequest);
  rpc Ensemble(EnsemblerRequest) returns (PredictValuesResponse);
}
```

Reply:
I see.. Then, is there a way to init the Fiber component with the UPI contract for the routes too? Or is that the problem you had summarised [here](https://github.com/gojek/fiber/pull/3#issuecomment-1230235098) and we will continue to use reflection for Fiber in all cases?

Reply:
For the [current implementation in fiber](https://github.com/gojek/fiber/blob/main/grpc/dispatcher.go#L108) no,

I believe it is do-able if we change it to do a search in the protoregistry (since its already registered in Turing) by providing the proto response type name.